plugins/pyembed/pyembed.cpp

```diff
@@ -19,11 +19,20 @@
 
 #ifdef _WIN32
 // There's an issue with Python redefining ssize_t resulting in errors - hide their definition
-#define ssize_t python_ssize_t
-#include "Python.h"
-#undef ssize_t
+  #define ssize_t python_ssize_t
+  #include "Python.h"
+  #undef ssize_t
 #else
-#include "Python.h"
+  #include "patchlevel.h"
+  #if (PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION >= 7)
+    #ifdef _POSIX_C_SOURCE
```

Comment:
Perhaps add a comment here such as "This Python version sets _POSIX_C_SOURCE and _XOPEN_SOURCE so we need to undef them here to prevent ...."